.github/workflows/build.yml

```diff
@@ -0,0 +1,41 @@
+name: Build HTML documentation
+
+on:
+  [push, pull_request]
+
+jobs:
+  build:
+    name: Build documentation
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - name: Install documentation dependencies
+        run: |
+          python -mpip install -r requirements.txt
+
+      - name: Install QuTiP from GitHub
+        run: |
+          python -mpip install git+git://github.com/qutip/qutip.git#egg=qutip[full]
+          python -c 'import qutip; qutip.about()'
+
+      - name: Build documentation
+        run: |
+          make html SPHINXOPTS="-W --keep-going -T"
+          # Above flags are:
+          #   -W : turn warnings into errors
+          #   --keep-going : do not stop after the first error
+          #   -T : display a full traceback if a Python exception occurs
+
+      - name: Upload built files
+        uses: actions/upload-artifact@v2
+        with:
+          name: qutip_html_docs
+          path: _build/html/*
+          if-no-files-found: error
```

Comment:
Does this mean that the compiled docs will be uploaded if the PR pass the tests but not merged? Shouldn't we only do this on push/merge?

Reply:
They're uploaded as an artifact, but not actually deployed anywhere.  It means that anyone looking at the PR can download the files and open them locally in a web browser to check that everything's rendered correctly.

I'm considering whether it's worth setting up a little system to deploy them to a bot-managed subdirectory in the `gh-pages` branch of a separate repository so they're autobuilt by GitHub Pages and available at (e.g.) `qutip.github.io/doc-preview/pr/<number>/<short commit>/index.html`, and bot-post the link in the comments.  Since the biggest "bugs" in documentation are usually stuff not rendering correctly, rather than the RST actually being _invalid_.

Reply:
> It means that anyone looking at the PR can download the files and open them locally in a web browser to check that everything's rendered correctly.

I see that indeed is useful. I hope there so no size limit on this.

Reply:
About the deployment, would we also consider ReadTheDocs? It will save a lot of time from our side. If we have the access to `qutip.org` domain, we can build a subdomain `qutip-docs/qutip.org` and direct the docs there.

Reply:
It's 2GB _per file_, if I recall correctly, and no limit on number.  If we ever hit that, we've got much larger problems.

About deployment: I meant temporary deployment, just as a preview while the PR is under review.  We probably shouldn't be exposing that to anything which links and may get indexed by a search enginge.